File: 34533.py
t = int(input())

def dfs(tree, cur):
    sub = []
    m = 0
    for nxt in tree[cur]:
        dep, string = dfs(tree, nxt)
        m = max(m, dep)
        sub.append((dep, string))
    sub.sort()

    res_string = "1"
    for dep, string in sub:
        res_string += string
    res_string += "0"
    return m + 1, res_string

def f():
    n = int(input())
    tree = [[] for _ in range(n)]

    line = list(map(int, input().split()))
    for i in range(1, n):
        tree[line[i] - 1].append(i)

    dep, string = dfs(tree, 0)
    print(string)

for _ in range(t):
    f()

File: 34533.cc
#include <bits/stdc++.h>
using namespace std;


pair<int, string> dfs(vector<vector<int>> tree, int cur) {
    vector<pair<int, string>> sub(0);
    for(int nxt : tree[cur]){
        sub.push_back(dfs(tree, cur));
    }

    
}

void f(){
    int n;
    cin >> n;

    vector<vector<int>> tree(n, vector<int>(0));

    int a;
    cin >> a;

    for(int i = 1; i<n; i++){
        int s;
        cin>> s;
        tree[s - 1] = i;
    }

    dfs(tree);
}

int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t;
    cin >> t;

    for(int i = 0; i<t; i++)
        f();
}